src/types.rs

```diff
@@ -349,6 +371,8 @@ impl LetsEncrypt {
 pub(crate) enum SigningAlgorithm {
     /// ECDSA using P-256 and SHA-256
     Es256,
+    /// HMAC with SHA-256,
```

Comment:
Given that RFC 8555 maintains a somewhat strict separation between MAC-based algorithms and other algorithms, I'd prefer a bit of stricter separation here. I'd propose an `enum JwsAlgorithm { Mac(MacAlgorithm), Signature(SignatureAlgorithm) }` for use in `Header`. Maybe then we could have `MacAlgorithm` and `SignatureAlgorithm` implement and have `Jwk` take a type implementing that trait as an argument?